explorer/interpreter/value.cpp

```diff
@@ -71,13 +71,12 @@ static auto GetMember(Nonnull<Arena*> arena, Nonnull<const Value*> v,
                  << "member " << f << " not in " << *witness;
         }
       }
-      case Value::Kind::SymbolicWitness: {
+      default:
+        CARBON_CHECK(isa<Witness>(witness))
```

Comment:
What's this doing versus the previous code?

IIUC, the previous code was using `default` for non-witness cases -- perhaps mainly to trap new, unhandled Witness cases.

However, `isa<Witness>` is redundant with respect to line 42 (`witness` is already a `Witness`), and should have no effect. If someone adds a new witness type, it's assumed to be a SymbolicWitness given the error on line 78.

I'm not sure that's the right implementation. A few thoughts:

- Was the old code more correct? Will there be Witness types that never support member lookup?
  - What's member lookup for a ConstraintWitness look like?
- Maybe the old switch setup was easier to read?
- Maybe delete this CHECK entirely if it's redundant? 
- Maybe if/else might work better than a switch with one case?
- Maybe "symbolic" should be replaced with something more dynamic?

Reply:
Thanks for calling this out. I think this was relevant with the old design for associated constants, where we might want to get a value here even if we have a symbolic witness, but it's not relevant any more and can be simplified.